aya/src/programs/perf_attach.rs

```diff
@@ -55,12 +54,12 @@ impl Link for PerfLink {
     type Id = PerfLinkId;
 
     fn id(&self) -> Self::Id {
-        PerfLinkId(self.perf_fd)
+        PerfLinkId(self.perf_fd.as_raw_fd())
     }
 
     fn detach(mut self) -> Result<(), ProgramError> {
-        let _ = perf_event_ioctl(self.perf_fd, PERF_EVENT_IOC_DISABLE, 0);
-        unsafe { close(self.perf_fd) };
+        let _ = perf_event_ioctl(self.perf_fd.as_fd(), PERF_EVENT_IOC_DISABLE, 0);
+        std::mem::drop(self.perf_fd);
```

Comment:
I am not sure if this is necessary, or if we should just let it drop on it's own when the scope ends. I just added the manual `drop` to match the exact same behavior as before where we closed the file descriptor immediately after the `PERF_EVENT_IOC_DISABLE`